### ISSUE TITLE
Test authentication work-flow

### ISSUE THREAD

**Issue by batpad** [mapbox/osmcha-frontend#1]

**Comment by ajithranka** [comment]
Elaborating on notes [here](https://github.com/willemarcel/osmcha-django/tree/restapi#how-to-login-using-the-oauth-api).

## Login

1. Make a POST request to `https://osmcha-django-api-test.tilestream.net/api/v1/login/social/token/openstreetmap`.

Returns

```
{
  "oauth_token_secret": "...",
  "oauth_token": "...",
  "oauth_callback_confirmed": "true"
}
```

2. Open a popup with the location set to `https://www.openstreetmap.org/oauth/authorize?oauth_token=<oauth_token>`.

3. On success, the popup redirects to `https://github.mapbox.io/osmcha-frontend/oauth-landing.html?oauth_token=<oauth_token>&oauth_verifier=<oauth_verifier>`. The `oauth-landing.html` page calls a method on the opener (parent window) passing `<oauth_token>` and `<oauth_verifier>` and closes the popup.

4. Make another POST request to `https://osmcha-django-api-test.tilestream.net/api/v1/login/social/token/openstreetmap?oauth_token=<oauth_token>&oauth_verifier=<oauth_verifier>` with the new tokens. Returns an API token.

5. Save the API token in the redux state and localStorage. Use for authenticated calls (such as fetching user details).

## Logout

Clear the API token from the redux state and localStorage. Reset the UI.

## Re-login

Check if the API token is saved in localStorage. If found, use for fetching user details.

## Fetching user details

POST request to `https://osmcha-django-api-test.tilestream.net/api/v1/users` with the API token as part of the header.

```
Authorization: Token ...
```

cc: @batpad @kepta @willemarcel

**Comment by ajithranka** [comment]
Edited Step 4 above. The `oauth_token` and `oauth_verifier` should be passed as part of the form data.

<img width="782" alt="screen shot 2017-05-08 at 5 02 02 pm" src="https://cloud.githubusercontent.com/assets/11095038/25802870/f3b15220-3411-11e7-8414-338b73d29ee7.png">

**Comment by kepta** [comment]
This has been implemented in #24 
Closing this.
